src/main/kotlin/com/autonomousapps/tasks/ComputeAdviceTask.kt

```diff
@@ -310,6 +401,27 @@ internal class DependencyAdviceBuilder(
         StandardTransform(coordinates, declarations, supportedSourceSets, buildPath, isKaptApplied).reduce(usages)
       }
   }
+
+  /**
+   * Returns whether this is a KMP non-common target that can just defer to a parent common declaration in
+   * [kmpCommonDeclarations].
+   *
+   * @see GradleVariantIdentification.kmpAttribute
+   */
+  private fun Coordinates.isKmpTargetThatShouldDeferToParent(
+    targetConfiguration: String?,
+    kmpCommonDeclarations: Map<String, Set<String>>
+  ): Boolean {
+    // This only applies to module coordinates.
+    if (this !is ModuleCoordinates) return false
+    if (targetConfiguration == null) return false
```

Comment:
As I indicated in a comment above, I find this null check a bit confusing out of context.

Reply:
[044613e](https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/919/commits/044613e44adcfd86a6f57abdb010bdbeaf596896)